File: app/src/main/java/com/sakurafish/pockettoushituryou/view/fragment/SettingsFragment.java
package com.sakurafish.pockettoushituryou.view.fragment;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;

import com.sakurafish.pockettoushituryou.R;
import com.sakurafish.pockettoushituryou.view.activity.WebViewActivity;

import de.psdev.licensesdialog.LicensesDialog;

public class SettingsFragment extends PreferenceFragment
        implements SharedPreferences.OnSharedPreferenceChangeListener {

    private Context mContext;

    public static SettingsFragment getInstance() {
        return new SettingsFragment();
    }

    @Override
    public void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mContext = getActivity();
        addPreferencesFromResource(R.xml.preferences);

        initLayout();
    }

    private void initLayout() {
        Preference credit = findPreference(getString(R.string.PREF_CREDIT));
        credit.setOnPreferenceClickListener(preference -> {
            showCreditDialog();
            return false;
        });

        Preference mailDev = findPreference(getString(R.string.PREF_MAIL_TO_DEV));
        mailDev.setOnPreferenceClickListener(preference -> {
            showMailIntent();
            return false;
        });

        Preference policy = findPreference(getString(R.string.PREF_PRIVACY_POLICY));
        policy.setOnPreferenceClickListener(preference -> {
            mContext.startActivity(WebViewActivity.createIntent(mContext
                    , "http://www.pockettoushituryou.com/static_pages/privacy_policy"
                    , getString(R.string.setting_privacy_policy)));
            return false;
        });

    }

    private void showCreditDialog() {
        new LicensesDialog.Builder(getActivity())
                .setNotices(R.raw.notices)
                .setIncludeOwnLicense(true)
                .build()
                .show();
    }

    private void showMailIntent() {
        Intent intent = new Intent(Intent.ACTION_SENDTO);
        intent.setData(Uri.parse("mailto:")); // only email apps should handle this
        intent.putExtra(Intent.EXTRA_EMAIL, new String[]{"sakurafish1@gmail.com"});
        intent.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.setting_mail_to_dev2));
        intent.putExtra(Intent.EXTRA_TEXT, getString(R.string.setting_mail_to_dev3));
        if (intent.resolveActivity(mContext.getPackageManager()) != null) {
            startActivity(intent);
        }
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
    }
}

File: app/src/main/java/com/sakurafish/pockettoushituryou/view/fragment/SettingsFragment.kt
package com.sakurafish.pockettoushituryou.view.fragment

import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.google.android.gms.oss.licenses.OssLicensesMenuActivity
import com.sakurafish.pockettoushituryou.BuildConfig
import com.sakurafish.pockettoushituryou.R
import com.sakurafish.pockettoushituryou.view.activity.WebViewActivity

class SettingsFragment : PreferenceFragmentCompat() {

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferences, rootKey)
        initLayout()
    }

    private fun initLayout() {

        val policy = findPreference(getString(R.string.PREF_PRIVACY_POLICY)) as Preference?
        policy?.setOnPreferenceClickListener {
            showPrivacyPolicy()
            true
        }
        val credit = findPreference(getString(R.string.PREF_CREDIT)) as Preference?
        credit?.setOnPreferenceClickListener {
            showCreditDialog()
            true
        }

        val mailDev = findPreference(getString(R.string.PREF_MAIL_TO_DEV)) as Preference?
        mailDev?.setOnPreferenceClickListener {
            showMailIntent()
            true
        }
    }

    private fun showPrivacyPolicy() {
        startActivity(WebViewActivity.createIntent(requireContext(), "file:///android_asset/www/privacy_policy.html", getString(R.string.setting_privacy_policy)))
    }

    private fun showCreditDialog() {
        startActivity(Intent(requireContext(), OssLicensesMenuActivity::class.java))
    }

    private fun showMailIntent() {
        val appVersion = BuildConfig.VERSION_NAME
        val manufacturer = Build.MANUFACTURER
        val model = Build.MODEL
        val device = if (model.startsWith(manufacturer)) {
            model
        } else {
            "$manufacturer $model"
        }
        val osVersion = android.os.Build.VERSION.SDK_INT
        val mailBody = requireContext().getString(R.string.setting_mail_to_dev_body, appVersion, device, osVersion)

        val intent = Intent(Intent.ACTION_SENDTO)
        intent.data = Uri.parse("mailto:") // only email apps should handle this
        intent.putExtra(Intent.EXTRA_EMAIL, arrayOf("sakurafish1@gmail.com"))
        intent.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.setting_mail_to_dev_subject))
        intent.putExtra(Intent.EXTRA_TEXT, mailBody)
        if (intent.resolveActivity(requireContext().packageManager) != null) {
            startActivity(intent)
        }
    }
}